File: readability/readability.py
from cs50 import get_string

texto = get_string('Insira o texto: ')

quant_palavras = 0
quant_letras = 0
quant_sentencas = 0
i = 0

tamanho = len(texto)

while i < tamanho:

    if texto[i].isalpha():
        quant_letras += 1

    if (i == 0 and texto[i] != " ") or (i != tamanho and texto[i] == " " and texto[i + 1] != " "):
        quant_palavras += 1

    if texto[i] == "." or texto[i] == "?" or texto[i] == "!":
        quant_sentencas += 1

    i += 1

L = (quant_letras / quant_palavras) * 100
S = (quant_sentencas / quant_palavras) * 100
indice = round(0.0588 * L - 0.296 * S - 15.8)

if indice < 1:
    print("Before Grade 1")
elif indice >= 16:
    print("Grade 16+")
else:
    print(f"Grade {indice}")

File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)

{
    int letters = 0;

    int words = 1; // não há um espaço vazio após a última palavra

    int sentences = 0;

    string text = get_string("Texto: ");

    int n = strlen(text);

    for (int i = 0; i < n; i++)

        if (text[i] >= 'a' && text[i] <= 'z')

            {
                letters += 1;
            }

        else if (text[i] >= 'A' && text[i] <= 'Z')

            {
                letters += 1;
            }

        else if (text[i] == ' ')

            {
                words += 1;
            }

        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')

            {
                sentences += 1;
            }

    float L = (letters * 100) / (float) words; // número médio de letras por 100 palavras

    float S = (sentences * 100) / (float) words; // número médio de sentenças por 100 palavras

    int col_liau_index = round(0.0588 * L - 0.296 * S - 15.8);

    if (col_liau_index < 1)

        printf("Before Grade 1\n");

    else if (col_liau_index <= 15)

        printf("Grade %i\n", col_liau_index);

    else

        printf("Grade 16+\n");

}